scripts/surface_high_engagement_issues.py

```diff
@@ -0,0 +1,503 @@
+                f" {TOKEN_REGEX.pattern!r}:"
+            ),
+            file=sys.stderr,
+        )
+        print(output, file=sys.stderr)
+        exit(1)
+
+    return match.groups()[0]
+
+
+def get_high_engagement_issues(headers: dict) -> list:
+    """
+    Fetch all high engagement issues from the repository.
+
+    Args:
+        headers (dict): HTTP headers for GitHub API requests.
+
+    Returns:
+        list: List of high engagement issues.
+    """
+    all_issues = []
+    page = 1
+    per_page = 100
+
+    while True:
+        url = f"https://api.github.com/repos/{GITHUB_REPO}/issues"
+        params = {
+            "state": "open",
+            "sort": "comments",
+            "direction": "desc",
+            "per_page": per_page,
+            "page": page,
+        }
+        response = httpx.get(url, headers=headers, params=params)
+        response.raise_for_status()
+        issues = response.json()
+
+        if not issues:
+            break
+
+        all_issues.extend(issues)
+        page += 1
+        print(f"Fetched page {page} with {len(issues)} issues")
+
+    high_engagement_issues = [
+        issue for issue in all_issues if issue["comments"] >= ENGAGEMENT_THRESHOLD
+    ]
+    print(f"Total high engagement issues: {len(high_engagement_issues)}")
+    return high_engagement_issues
+
+
+def issue_has_new_comment(issue, new_comment_interval_days, headers: dict) -> bool:
+    """
+    Check if an issue has a new comment within the specified interval.
+
+    Args:
+        issue (dict): The issue to check.
+        new_comment_interval_days (int): Interval in days to check for new comments.
+        headers (dict): HTTP headers for GitHub API requests.
+
+    Returns:
+        bool: True if there is a new comment within the interval, False otherwise.
+    """
+    comments_url = issue["comments_url"]
+    response = httpx.get(comments_url, headers=headers)
+    response.raise_for_status()
+    comments = response.json()
+
+    if comments:
+        latest_comment = max(comments, key=lambda comment: comment["created_at"])
+        latest_comment_date = datetime.strptime(
+            latest_comment["created_at"], "%Y-%m-%dT%H:%M:%SZ"
+        )
+        if latest_comment_date > datetime.utcnow() - timedelta(
+            days=new_comment_interval_days
+        ):
+            return True
+    return False
+
+
+def prioritized_recently(issue_number: int, headers: dict) -> bool:
+    """
+    An issue is considered to have been prioritized recently if it was removed within the last LABEL_REMOVAL_INTERVAL_MONTHS months.
+    We don't want to re-add the "Needs Priority" label if it was removed recently as there is likely to be further discussion
+    on this issue right after the issue is prioritized and the label is removed.
+
+    It's possible we may want to increase the LABEL_REMOVAL_INTERVAL_MONTHS to a larger value if we find that the label is being
+    re-added too frequently.
```

Comment:
Would appreciate thoughts on this

Reply:
hmm yeah it is a complicated thing to generalize - I think this is a good starting place!